### PR TITLE
No finally syntax extension

### PR DESCRIPTION
**Pull request by UnixJunkie**

### CODE REVIEW

**Comment by UnixJunkie** [comment]
Since I'm new to the code base, I try to do steps by steps.
Martin, please tell me if you don't have time to review PRs.

File: main.ml [hunk 1]
```diff
@@ -346,9 +346,13 @@ let split_file =
...
-  finally (* comment out for debugging: *)
-    Pipeline.remove meta_name;
-    Pipeline.remove prog_name
+  with exn ->
+    begin (* finally *)
+      (* comment out for debugging: *)
+      Pipeline.remove meta_name;
+      Pipeline.remove prog_name;
+      raise exn
+    end
```
Comment (mjambon) [inline main.ml:355]: The "finally" block must be called also if no exception is raised. You should use `Fun.protect` for this. See https://caml.inria.fr/pub/docs/manual-ocaml/libref/Fun.html

**Comment by mjambon** [comment]
> Martin, please tell me if you don't have time to review PRs.

It depends how easy it is for me. I don't really want to be involved in new design decisions but I can help figure out old stuff.

File: main.ml (hunk 1 above)
Comment (UnixJunkie) [reply main.ml:355]: Ok, I will use that. But then it wil add a constraint (ocaml >= 4.08).

Comment (mjambon) [reply main.ml:355]: Sounds reasonable. Otherwise you can define your own `protect` function.

Comment (mjambon) [reply main.ml:355]: Something like this:
```ocaml
let protect ~finally f =
  try let res = f () in
  finally ();
  res
with e ->
  finally ();
  raise e
```

```
# protect ~finally:(fun () -> print_endline "bye") (fun () -> print_endline "hello"; raise Exit);;
hello
bye
Exception: Stdlib.Exit.
```

**Comment by UnixJunkie** [comment]
I removed the trailing whitespaces in one of the files, sorry this is polluting the diff a little.

**Pull request merged**